src/main.cpp

```diff
@@ -655,6 +662,9 @@ CBlockIndex* FindForkInGlobalIndex(const CChain& chain, const CBlockLocator& loc
             CBlockIndex* pindex = (*mi).second;
             if (chain.Contains(pindex))
                 return pindex;
+            if (pindex->GetAncestor(chain.Height()) == chain.Tip()) {
+                return chain.Tip();
```

Comment:
I understand this as: if we have a block hash that the caller sent us in our `mapBlockIndex` that is not in the main chain, but is a descendant of our best block, then send our best block. Is there a case where our `mapBlockIndex` knows about a block that is higher than the tip of our chain? Are these data structures not kept in sync so that all the blocks in `mapBlockIndex` are also in `chain`?